src/ResponseBuilder.cpp

```diff
@@ -102,7 +102,10 @@ void ResponseBuilder::appendHeaders(const HTTPRequest& request)
 {
 	if (!m_responseBody.empty()) {
 	// Content-Type
-		m_responseHeader << "Content-Type: " << getMIMEType(webutils::getFileExtension(request.targetResource)) << "\r\n";
+		if (m_responseBody.find('{') != std::string::npos)
+			m_responseHeader << "Content-Type: application/json\r\n";
```

Comment:
This makes every file with a curly bracket a json file.

I would change m_request.targetResource to "created.json" or "deleted.json" and add a MIME mapping for .json.
The target resource is not get sent back, so we can reuse it. See for example setDefaultErrorPage() which sets it to "error.html" to get html mapping

Reply:
This would also not be correct as this would require/ infer that all POST requests have a json body which is not necessarily the case. The json content-type only refers to the response body which is in the json format. The target resource can have any format.

It would be necessary to find a distinctive feature in the json format in order to recognise it. As we only have responses in either HTML or json, I chose the bracket. Alternatively one could always use a content-type json whenever a successful POST or DELETE response is sent.

Reply:
This adds the header for _our response_. 

What came in as POST is already processed, the content-type was given to us by the browser. We send our response, which in case of POST and DELETE right now is always in .json (from FileWriteHandler, DeleteHandler) or in case of GET can be .html, .txt, .png, whatever.

The Bracket '{' or byte 0x7B is far from distinct. Our index.html has brackets for styling. If you start server and navigate to index.html, the browser will give you a .json representation and will not render html (since we say its json). I think the possibility of a binary format like png having this byte is also high. 
MIME mapping is very "dumb" in that in only cares for file extension and not magic numbers or stuff.
Since the target resource for POST and DELETE is not used anymore at this stage (we already created / deleted the file) we can just reuse it to identify our response as json

Reply:
To my understanding, what you're asking to do is to set the target resource in the request to:

E.g. /uploads/myfile.json

POST /uploads/myfile.json HTTP/1.1
...

Then define a mime_mapping element which recognises the .json extension and then set the Content-Type header in the response to application/json.

But this would imply that the posted content is in json format, although it can be anything.

If you mean sth else, please illustrate, best with an example.

Reply:
No. After successfully processing the DELETE request, reuse our variable targetResource and set it to "deleted.json". Like so:
```
if (m_request.method == MethodDelete) {
  DeleteHandler deleteHandler(m_fileSystemPolicy);
  m_responseBody = deleteHandler.execute(m_request.targetResource, m_request.httpStatus);
  if (m_responseBody.empty())
    handleErrorBody();
  m_request.targetResource = "deleted.json"; <--- here we reuse variable
}
```

Example: existing implementation of ResponseBodyHandler::setDefaultErrorPage():
```
void ResponseBodyHandler::setDefaultErrorPage()
{
	m_responseBody = getDefaultErrorPage(m_request.httpStatus);
	m_request.targetResource = "error.html";
}
```

When we return to ResponseBuilder, the target resource has a json extension. If you add a mapping for it, then the response would have the correct content-type: json

Reply:
Achsooo, ja that can be done

Reply:
Implemented